components/button/button-subtle.js

```diff
@@ -50,7 +50,9 @@ class ButtonSubtle extends ButtonMixin(RtlMixin(LitElement)) {
 			 * REQUIRED: Text for the button
 			 * @type {string}
 			 */
-			text: { type: String, reflect: true }
+			text: { type: String, reflect: true },
+			_hasCustomIcon: { state: true },
+			_hasIcon: { type: Boolean, reflect: true, attribute: '_has-icon' }
```

Comment:
I always feel a bit uncomfortable when we sprout private properties as attributes on the host. I know there are cases where we've done it, particularly when the component doesn't have any intermediate element in the shadowDOM suitable for applying the class, such as when rendering is being done entirely by a mixin. I sort of view the host as being the public API to the consumer, as it's in the light DOM and all the other standard public attributes are on it. Do others agree? To avoid this, a `has-icon` class could be rendered internally on the `button`, thereby keeping this buried in the shadowDOM.

Reply:
Agreed, I also don't love it. Good idea, I'll work on switching it out.